backend/inspirehep/records/marshmallow/literature/bibtex.py

```diff
@@ -277,7 +278,12 @@ def hyphenate_if_possible(no_hyphens):
 
     @pre_dump
     def filter_data(self, data):
-        control_number = str(data.get("control_number"))
-        data["doc_type"] = BibTexCommonSchema.get_bibtex_document_type(data)
-        data["texkey"] = get_value(data, "texkeys[0]", default=control_number)
-        return data
+        processed_data = deepcopy(data)
```

Comment:
do you really need the **deep**copy here?

Reply:
I'm actually not sure how to do this. So the data is being mutated here as well with adding ```texkey``` and ```doc_type``` which are being used by the other methods. Another idea would be to ditch the pre_dump and have a method to get those 2 fields whenever we need them without adding them to the data?

Reply:
I think it's an ordering issue: other methods need the `doc_type` to decide how to serialize other fields, but there's no way to enforce the order in which they get evaluated. The `texkey` though looks like it could be moved out to a regular field. In any case, `copy` is sufficient, you don't need to `deepcopy` because you're only modifying the `data` itself (by adding a key), not nested fields.